varats/varats/data/reports/llvm_coverage_report.py

```diff
@@ -0,0 +1,708 @@
+                    region.childs.sort()
+                    for child in childs_to_move:
+                        child.parent = region
+                        node.childs.remove(child)
+                node.childs.append(region)
+                node.childs.sort()
+                region.parent = node
+                # Actually this is possible,
+                # e.g. a for loop can be executed
+                # more often than its function.
+                #assert region.count <= node.count
+                break
+
+    def merge(self, region: CodeRegion) -> None:
+        """Merges region into self by adding all counts of region to the counts
+        of self."""
+        for x, y in zip(self.iter_breadth_first(), region.iter_breadth_first()):
+            assert x == y, "CodeRegions are not identical"
+            x.count += y.count
+
+    def diff(self, region: CodeRegion) -> None:
+        """Builds the difference between self (base code) and region (new code)
+        by comparing them."""
+        for x, y in zip(self.iter_breadth_first(), region.iter_breadth_first()):
+            assert x == y, "CodeRegions are not identical"
+            if x.is_covered() and y.is_covered(
+            ) or not x.is_covered() and not y.is_covered():
+                # No difference in coverage
+                x.count = 0
+            elif x.is_covered() and not y.is_covered():
+                # Coverage decreased
+                x.count = -1
+            elif not x.is_covered() and y.is_covered():
+                # Coverage increased
+                x.count = 1
+
+    def is_identical(self, other: object) -> bool:
+        """Is the code region equal and has the same coverage?"""
+        if not isinstance(other, CodeRegion):
+            return False
+        if not (self == other and self.count == other.count):
+            return False
+        for code_region_a, code_region_b in zip(
+            self.iter_breadth_first(), other.iter_breadth_first()
+        ):
+            if not (
+                code_region_a == code_region_b and
+                code_region_a.count == code_region_b.count
+            ):
+                return False
+        return True
+
+    # Compare regions only depending on their
+    # start lines and columns + their type
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, CodeRegion):
+            return False
+        return (
+            self.start.line == other.start.line and
+            self.start.column == other.start.column and
+            self.end.line == other.end.line and
+            self.end.column == other.end.column and self.kind == other.kind
+        )
+
+    def __lt__(self, other: CodeRegion) -> bool:
+        if (
+            self.start.line < other.start.line or
+            self.start.line == other.start.line and
+            self.start.column < other.start.column
+        ):
+            return True
+        return False
```

Comment:
```suggestion
            return True

        return False
```
I personally, like a few empty lines here and there to separate logical blocks, otherwise it's harder to spot for the eye